src/GitHub.App/Services/GistCreator.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Reactive.Linq;
+using System.Reactive.Threading.Tasks;
+using System.Threading.Tasks;
+using GitHub.Api;
+using GitHub.Extensions;
+using GitHub.Models;
+using GitHub.UI;
+using Octokit;
+using GitHub.Factories;
+using GitHub.Primitives;
+
+namespace GitHub.Services
+{
+    [Export(typeof(IGistCreator))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class GistCreator : IGistCreator
+    {
+        readonly ISelectedTextProvider selectedTextProvider;
+        readonly IConnectionManager connectionManager;
+        readonly IRepositoryHosts repositoryHosts;
+        readonly IUIProvider uiProvider;
+        readonly IApiClient apiClient;
+
+        [ImportingConstructor]
+        public GistCreator(ISelectedTextProvider selectedTextProvider, IConnectionManager connectionManager, 
+            IRepositoryHosts repositoryHosts, IUIProvider uiProvider, IApiClientFactory apiClientFactory)
+        {
+            this.selectedTextProvider = selectedTextProvider;
+            this.connectionManager = connectionManager;
+            this.repositoryHosts = repositoryHosts;
+            this.uiProvider = uiProvider;
+            this.apiClient = apiClientFactory.Create(HostAddress.GitHubDotComHostAddress);
+        }
+
+        public async Task<Gist> CreateFromSelectedText()
+        {
+            await EnsureLoggedIn();
+
+            var selectedText = selectedTextProvider.GetSelectedText().ToTask().Result;
+
+            // Show the Create Gist Window, and on Create button clicked execute the following
+            return await apiClient.CreateGist("NameWillBeEnteredInThePopup", true, selectedText)
```

Comment:
Actually, there's little reason to return a `Task<Gist>` here instead of the original `IObservable<Gist>`. The only reason certain APIs deal with `Task<T>` instead of `IObservable<T>` is because instances of classes created in the Team Explorer home page are constrained to load as few dependencies as possible, and that means no reactive extensions. That's not an issue for this, since this loads on demand only when a user requests a gist to be created, so we can use rx here just fine.